ra_aid/callbacks/default_callback_handler.py

```
@@ -281,9 +350,22 @@ def _update_token_counts(self, token_usage: dict, duration: float) -> None:
 
-            # Calculate costs using Decimal arithmetic
-            input_cost = Decimal(prompt_tokens) * self.input_cost_per_token
-            output_cost = Decimal(completion_tokens) * self.output_cost_per_token
+            # Determine cost per token for this specific call, considering tiers
+            current_input_cost_per_token = self.input_cost_per_token
+            current_output_cost_per_token = self.output_cost_per_token
+
+            if self.tiered_costs:
+                threshold = self.tiered_costs.get("tier_threshold", 0)
+                if prompt_tokens > threshold:
+                    current_input_cost_per_token = self.tiered_costs.get("input_over_200k", current_input_cost_per_token)
```

Pylint (3.3.7)

[convention] Line too long (121/100)

(C0301)